app/Http/Controllers/CampaignRegistrationController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Campaign;
+use App\Models\Character;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+
+class CampaignRegistrationController extends Controller
+{
+    /**
+     * @param \Illuminate\Http\Request $request
+     * @return \Inertia\Response
+     */
+    public function create(Request $request)
+    {
+        $characters = Auth::user()->characters;
+
+        $code = $request->validate([
+            'code' => "sometimes|string"
+        ])['code'] ?? null;
+
+        $campaign = Campaign::with('characters', 'users')->where('code', $code)->first();
+
+        return Inertia::render('Campaign/Detail/CampaignDetail', compact('characters', 'campaign'));
```

Comment:
it'd also pass down code here, will be convenient for the front-ends later
```suggestion
        return Inertia::render('Campaign/Detail/CampaignDetail', compact('characters', 'campaign', 'code'));
```